src/audio/codec_adapter/codec/waves.c

```diff
@@ -0,0 +1,726 @@
+			src_fmt->channels);
+		return -EIO;
+	}
+
+	comp_dbg(dev, "waves_effect_check() done");
+	return 0;
+}
+
+static int waves_effect_init(struct comp_dev *dev)
+{
+	/* initializes MaxxEffect based on stream parameters
+	 * returnes non-zero value in case of error
+	 */
+	struct codec_data *codec = comp_get_codec(dev);
+	struct waves_codec_data *waves_codec = codec->private;
+	struct comp_data *component = comp_get_drvdata(dev);
+
+	const struct audio_stream *src_fmt = &component->ca_source->stream;
+
+	MaxxStatus_t status;
+	MaxxBuffer_Format_t sample_format;
+	MaxxBuffer_Layout_t buffer_format;
+	int32_t sample_bytes;
+	MaxxStreamFormat_t *i_formats[NUM_IO_STREAMS] = { &waves_codec->i_format };
+	MaxxStreamFormat_t *o_formats[NUM_IO_STREAMS] = { &waves_codec->o_format };
+
+	comp_dbg(dev, "waves_effect_init() start");
+
+	sample_format = sample_format_convert_sof_to_me(src_fmt->frame_fmt);
+	if (sample_format < 0) {
+		comp_err(dev, "waves_effect_init() sof sample format %d not supported",
+			src_fmt->frame_fmt);
+		return -EIO;
+	}
+
+	buffer_format = buffer_format_convert_sof_to_me(component->ca_source->buffer_fmt);
+	if (buffer_format < 0) {
+		comp_err(dev, "waves_effect_init() sof buffer format %d not supported",
+			component->ca_source->buffer_fmt);
+		return -EIO;
+	}
+
+	sample_bytes = sample_format_convert_to_bytes(sample_format);
+	if (sample_bytes < 0) {
+		comp_err(dev, "waves_effect_init() sample_format %d not supported",
+			sample_format);
+		return -EIO;
+	}
+
+	waves_codec->request_max_bytes = 0;
+	waves_codec->response_max_bytes = 0;
+	waves_codec->response = 0;
+	waves_codec->i_buffer = 0;
+	waves_codec->o_buffer = 0;
+
+	waves_codec->i_format.sampleRate = src_fmt->rate;
+	waves_codec->i_format.numChannels = src_fmt->channels;
+	waves_codec->i_format.samplesFormat = sample_format;
+	waves_codec->i_format.samplesLayout = buffer_format;
+
+	waves_codec->o_format = waves_codec->i_format;
+
+	waves_codec->sample_size_in_bytes = sample_bytes;
+	waves_codec->buffer_samples = src_fmt->rate * 2 / 1000; /* 2 ms io buffers */
+	waves_codec->buffer_bytes = waves_codec->buffer_samples * src_fmt->channels *
+		waves_codec->sample_size_in_bytes;
+
+	comp_info(dev, "waves_effect_init() rate %d, channels %d",
+		waves_codec->i_format.sampleRate,
+		waves_codec->i_format.numChannels);
+
+	comp_info(dev, "waves_effect_init() format %d, layout %d, frame %d",
+		waves_codec->i_format.samplesFormat,
+		waves_codec->i_format.samplesLayout,
+		waves_codec->buffer_samples);
+
+	status = MaxxEffect_Initialize(waves_codec->effect, i_formats, 1, o_formats, 1);
+
+	if (status) {
+		comp_err(dev, "waves_effect_init() MaxxEffect_Initialize() error %d",
+			status);
+		return -EIO;
+	}
+
+	comp_dbg(dev, "waves_effect_init() done");
+	return 0;
+}
+
+#define R32(n) (\
+	(((n) & 0x000000FF) << 24)| \
+	(((n) & 0x0000FF00) << 8) | \
+	(((n) & 0x00FF0000) >> 8) | \
+	(((n) & 0xFF000000) >> 24))
+
+#define DUMP_REVISION_ITERATION(p, l) do {\
```

Comment:
and this looks like a general purpose hexdump ? These are probably useful in a core API so that others can also use them.

Reply:
not sure it will be useful for others.
It is here as a workaround only until we will have "get" requests from codec_adapter